soc/realtek/ec/common/rts5912_imgtool/img_gen.py

```diff
@@ -0,0 +1,104 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2024 Realtek Semiconductor Corporation, SIBG-SD7
+# Author: Dylan Hsieh <dylan.hsieh@realtek.com>
+
+"""
+The RTS5912 specific image header shows the bootROM how to
+load the image from flash to internal SRAM, this script obtains
+the header to original BIN and output a new BIN file.
+"""
+
+import struct
+import os
+import click
+
+IMAGE_MAGIC = 0x524C544B    # ASCII 'RLTK'
+IMAGE_HDR_SIZE = 32
+RAM_LOAD = 0x0000020
+
+class BasedIntParamType(click.ParamType):
+    """
+    Class converting the string to integer with literals prefixed
+    with bin/oct/hex format
+    """
+    name = 'integer'
+    def convert(self, value, param, ctx):
+        try:
+            return int(value, 0)
+        except ValueError:
+            self.fail('Please use code literals prefixed with 0b/0B, 0o/0O, or 0x/0X as necessary.')
+        return None
+
+@click.command()
+@click.option('-L', '--load-addr', type=BasedIntParamType(), required=False,
+              help='Load address for image when it should run from RAM.')
+@click.option('--spi-freq', type=click.Choice(['0', '1', '2', '3']),
+              required=False, default='3',
+              help='Specify the frequency of SPI I/F.')
+@click.option('--spi-rdcmd', type=click.Choice(['0x03', '0x0B', '0x3B', '0x6B']),
+              required=False, default='0x03',
+              help='Specify the command for flash read.')
+@click.argument('original_bin')
+@click.argument('signed_bin')
```

Comment:
please use plain argparse, as everywhere else in the repo

Reply:
We have changed the code to use plain argparse here.